src/charm.py

```diff
@@ -104,7 +106,30 @@ def _configure_hacluster(self):
         else:
             self.hacluster.remove_service("nginx", "nginx")
 
-    def _configure_nginx(self, servers: Dict[int, Set]):
+    def _configure_nginx(self):
+        """Configure NGINX based on provided directives.
+
+        This method configures main and events contexts with the directives provided
+        in the configuration, removes the default NGINX site, and writes the NGINX log
+        rotation configuration.
+        """
+        context = {}
+        try:
+            context["main"] = yaml.safe_load(self.config.get("main-context"))
+            context["events"] = yaml.safe_load(self.config.get("events-context"))
+        except YAMLError:
```

Comment:
maybe log the yaml error so someone sees their mistake when looking at the debug log

```suggestion
        except YAMLError:
            log.exception("Encountered juju config parsing error")
```